var/spack/repos/builtin/packages/llvm-doe/package.py

```diff
@@ -0,0 +1,525 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os.path
+import re
+import sys
+
+import llnl.util.tty as tty
+import spack.util.executable
+
+
+class LlvmDoe(CMakePackage, CudaPackage):
+    """This package provides a collection of the experimental LLVM projects done
+    by the US DOE research and development teams.
+    """
+
+    homepage = "https://github.com/llvm-doe-org"
+    url = "https://github.com/llvm-doe-org/llvm-project/archive/llvmorg-10.0.0.zip"
+    git = "https://github.com/llvm-doe-org/llvm-project"
+    maintainers = ['shintaro-iwasaki']
+
+    family = "compiler"  # Used by lmod
+
+    version('main', branch='doe', preferred=True)
```

Comment:
Usually we name versions the same as the branches they refer to, but I know `master` etc. have special semantics for ordering. @adamjstewart What would be the way to go here?

Reply:
Thank you.  To avoid confusing and remove `preferred=True`, I changed `main` to `master` (and by doing so the `doe` branch becomes the default branch). `doe` is the branch name of the "default/main/master" branch of this repository.  If this naming is inappropriate, please let me know.

(As far as I checked `main` works as the default branch if `preferred=True` is set.  It seems that `main` has no meaning at this point.  I used `main` instead of `master` for a more inclusive branch name, but we should have followed the current naming rule since using `master` can avoid confusion.)

Reply:
To be clear, I don't mind calling the branch in any way that seems most appropriate to you as a maintainer. What I meant is that for versions tied to a branch most of the packages do:
```python
version('XYZ', branch='XYZ')
```
so I was just curious why the version was not `doe` in this particular case. Using `main`, `master`, `develop`, etc. triggers special semantics for comparison, as these names are treated like infinity and they compare greater as any other non-infinity version. See:

https://github.com/spack/spack/blob/de3504ca2965e8cdba4e615d6bcb590f274aebef/lib/spack/spack/version.py#L42-L43

and the associated Python module.

Reply:
Oh I got it. Thank you. I changed it to the following so that the version name is the same as the branch name.
```py
    version('doe', branch='doe', preferred=True)
```